roles/ceph-client/tasks/create_users_keys.yml

```diff
@@ -12,6 +12,19 @@
   when:
     - keys_tmp is defined
 
+# dummy container setup is only supported on x86_64
+# when running with containerized_deployment: true this task
+# creates a group that contains only x86_64 hosts.
+# when running with containerized_deployment: false this task
+# will add all client hosts to the group (and not filter).
+- name: create filtered clients group
+  add_host:
+    name: "{{ item }}"
+    groups: _filtered_clients
+  with_items: "{{ groups[client_group_name] }}"
+  when:
+    - (hostvars[item]['ansible_architecture'] == 'x86_64') or (not containerized_deployment)
```

Comment:
Shouldn't this be?

```
when:
  - containerized_deployment
  - hostvars[item]['ansible_architecture'] == 'x86_64'
```

Reply:
@leseb - i think that would create a situation where on containerized_deployment: false that the filtered_clients group contains no hosts, we want the opposite which is when containerized_deployment is false or its an x86_64 host, then add it to the group.

Reply:
Good point, can't we then do something like `groups.get('_filtered_clients', 'client_group_name')`? So if `_filtered_clients` does not exist we use `client_group_name`?

Reply:
We could - but it makes line 130 difficult because we reference host [0] in the group, rather than using groups.get.